services/hotelsService.js

```diff
@@ -98,11 +103,34 @@ exports.getHotelIdsForOwner = async (hotelOwnerId) => {
 }
 
 exports.getHotelOwnerId = async (hotelId) => {
-  const result = await Hotel.findOne({ _id: hotelId }).select('ownerId -_id')
+  const hotel = await Hotel.findOne({ _id: hotelId }).select('ownerId -_id')
 
-  if (!result) {
+  if (!hotel) {
     return null
   }
 
-  return result.ownerId
+  return hotel.ownerId
 }
+
+exports.getAvailableHotelRooms = async (req) => {
+  const { id: hotelId } = req.params
+  const { adults, children, startDate, endDate } = req.query
+
+  const hotel = await Hotel.findOne({ _id: hotelId })
+  
+  if(!hotel) {
+    throw new NotFoundError('Hotel not found.')
+  }
+
+  const rooms = hotel.rooms.filter(r => r.beds.single + r.beds.double * 2 >= parseInt(adults) + parseInt(children))
```

Comment:
I think such code was already used earlier, so maybe refactor this (maybe whole function) to another function and use it in both places

Reply:
There is getGuestInRoom but it wont work here. Creating separate function instead of one line makes no sense in this particular example

Reply:
Or maybe it makes sense.

Reply:
Of course it make sense and here I marked this line only as an example, but earlier was used also the loop below so not only one line is repeated

Reply:
loop is different